client/python/client.py

```diff
@@ -0,0 +1,284 @@
+            job_ids=job_ids, job_set_id=job_set_id, queue=queue,
+            new_priority=new_priority
+        )
+        response = self.submit_stub.ReprioritizeJobs(request)
+        return response
+
+    def create_queue(self, name, **queue_params):
+        request = submit_pb2.Queue(name=name, **queue_params)
+        response = self.submit_stub.CreateQueue(request)
+        return response
+
+    def update_queue(self, name, **queue_params):
+        request = submit_pb2.Queue(name=name, **queue_params)
+        response = self.submit_stub.UpdateQueue(request)
+        return response
+
+    def delete_queue(self, name):
+        request = submit_pb2.QueueDeleteRequest(name=name)
+        response = self.submit_stub.DeleteQueue(request)
+        return response
+
+    def get_queue(self, name):
+        request = submit_pb2.QueueGetRequest(name=name)
+        response = self.submit_stub.GetQueue(request)
+        return response
+
+    def get_queue_info(self, name):
+        request = submit_pb2.QueueInfoRequest(name=name)
+        response = self.submit_stub.GetQueueInfo(request)
+        return response
+
+    def watch_events(self, on_event, queue, job_set_id, from_message_id=None):
+        jsr = event_pb2.JobSetRequest(queue=queue, id=job_set_id, from_message_id=from_message_id, watch=True, errorIfMissing=True)
+        event_stream = self.event_stub.GetJobSetEvents(jsr)
+        def thread_func():
+            try:
+                nonlocal event_stream
+                for event in event_stream:
+                    on_event(event)
+            except grpc._channel._MultiThreadedRendezvous as e:
+                if e.code() == grpc.StatusCode.CANCELLED:
+                    pass
+                # process cancelled status
+                elif e.code() == grpc.StatusCode.UNAVAILABLE and 'Connection reset by peer' in e.details():
+                    pass
+                # process unavailable status
+                else:
+                    raise
+
+        th = threading.Thread(target=thread_func)
+        th.start()
+        return event_stream
+
+    def unwatch_events(self, event_stream):
+        event_stream.cancel()
+
+class ArmadaClientTest:
+    def __init__(self, host, port):
+        # TODO: generalize this so tests can be run with a variety of auth schemas
+
+        basic_auth_data = AuthData(AuthMethod.Basic, username='testuser', password='asdfasdf')
+        self.client = ArmadaClient(host, port, basic_auth_data, testing=True)
+
+    # private static ApiJobSubmitRequest CreateJobRequest(string jobSet)
+    def job_submit_request_items_for_test(self, queue, job_set_id):
+        pod = core_v1.PodSpec(
+            volumes=[
+                core_v1.Volume(
+                    name="root-dir",
+                    volumeSource=
+                        core_v1.VolumeSource(flexVolume=
+                            core_v1.FlexVolumeSource(
+                                driver="gr/cifs",
```

Comment:
I think this is a GR-specific component that shouldn't be in the public repo.

Reply:
I got that from the C# example. With what should it be replaced?

Reply:
@severinson Can probably correct me but this looks like to me like it is generating a static definition for a pod.  You can probably just use really simple pod definitions or reference an existing one in examples.